tests/phpunit/tests/link/getAdjacentPost.php

```diff
@@ -350,4 +350,82 @@ public function filter_excluded_terms( $excluded_terms ) {
 		$excluded_terms[] = $this->exclude_term;
 		return $excluded_terms;
 	}
+
+	/**
+	 * @ticket 41131
+	 */
+	public function test_get_adjacent_post_cache() {
+		// Need some sample posts to test adjacency.
+		$post_one = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'First',
+				'post_date'  => '2012-01-01 12:00:00',
+			)
+		);
+
+		$post_two = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'Second',
+				'post_date'  => '2012-02-01 12:00:00',
+			)
+		);
+
+		$post_three = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'Third',
+				'post_date'  => '2012-03-01 12:00:00',
+			)
+		);
+
+		$post_four = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'Fourth',
+				'post_date'  => '2012-04-01 12:00:00',
+			)
+		);
+
+		// Assign some terms.
+		wp_set_object_terms( $post_one->ID, 'WordPress', 'category', false );
+		wp_set_object_terms( $post_three->ID, 'WordPress', 'category', false );
+
+		wp_set_object_terms( $post_two->ID, 'plugins', 'post_tag', false );
+		wp_set_object_terms( $post_four->ID, 'plugins', 'post_tag', false );
+
+		// Test normal post adjacency.
+		$this->go_to( get_permalink( $post_two->ID ) );
+
+		// Test getting the right result
+		$first_run = get_adjacent_post( false, '', true );
+		$this->assertEquals( $post_one, $first_run, 'Get first post when on second post' );
+		$this->assertNotEquals( $post_two, $first_run, 'Do not get second post when on second post' );
```

Comment:
Thanks for adding `$message` as per https://github.com/WordPress/wordpress-develop/pull/3366#discussion_r1050298334 feedback. The `$message` parameter should explain what went wrong in the test. This message will be displayed when the assertion fails and can help immensely when debugging a test.